homeassistant/components/mobile_app/webhook.py

```diff
@@ -307,6 +312,33 @@ def wrapper(func):
     )
 
 
+@WEBHOOK_COMMANDS.register("enable_encryption")
+async def webhook_enable_encryption(hass, config_entry, data):
+    """Handle a encryption enable webhook."""
+    if config_entry.data[ATTR_SUPPORTS_ENCRYPTION]:
+        _LOGGER.warning(
+            "Refusing to enable encryption for %s because it is already enabled!",
+            config_entry.data[ATTR_DEVICE_NAME],
+        )
+        return error_response(
+            ERR_ENCRYPTION_ALREADY_ENABLED, "Encryption already enabled"
+        )
+
+    if not supports_encryption():
+        _LOGGER.warning(
+            "Unable to enable encryption for %s because libsodium is unavailable!",
+            config_entry.data[ATTR_DEVICE_NAME],
+        )
+        return error_response(ERR_ENCRYPTION_NOT_AVAILABLE, "Encryption is unavailable")
+
+    config_entry.data[ATTR_SUPPORTS_ENCRYPTION] = True
```

Comment:
We should not write directly to config_entry.data. Instead, write it to an object and then pass it to `async_updat_entry. 

```python
data = {
  **config_entry.data,
  ATTR_SUPPORTS_ENCRYPTION: True,
  CONF_SECRET: secrets.token_hex(SecretBox.KEY_SIZE)
}
```